cfme/tests/cli/test_appliance_console.py

```diff
@@ -0,0 +1,158 @@
+import pytest
+from collections import namedtuple
+from wait_for import wait_for
+from utils import version
+from utils.log_validator import LogValidator
+
+TimedCommand = namedtuple('TimedCommand', ['command', 'timeout'])
+
+
+@pytest.mark.smoke
+def test_black_console(request, appliance):
+    command_set = ('exec > >(tee /tmp/opt.txt)', 'ap', '')
+    appliance.appliance_console.run_commands(command_set)
+    assert appliance.ssh_client.run_command("cat /tmp/opt.txt | grep 'CFME Virtual Appliance'")
+    assert appliance.ssh_client.run_command("cat /tmp/opt.txt | grep 'CFME Server:'")
+    assert appliance.ssh_client.run_command("cat /tmp/opt.txt | grep 'CFME Database:'")
+    assert appliance.ssh_client.run_command("cat /tmp/opt.txt | grep 'CFME Version:'")
+
+
+def test_black_console_set_hostname(request, appliance):
+    hostname = 'Elite-QE.redhat.com'
+    if appliance.version >= "5.8":
+        command_set = ('ap', '', '1', '4', hostname)
+    else:
+        command_set = ('ap', '', '4', hostname)
+    appliance.appliance_console.run_commands(command_set)
+
+    def is_hostname_set(appliance):
+        assert appliance.ssh_client.run_command("hostname -f | grep {hostname}"
+            .format(hostname=hostname))
+    wait_for(is_hostname_set, func_args=[appliance])
+    return_code, output = appliance.ssh_client.run_command("hostname -f")
+    assert output.strip() == hostname
+    assert return_code == 0
+
+
+def test_black_console_internal_db(request, app_creds, temp_appliance_unconfig_funcscope):
+    pwd = app_creds['password']
+    if temp_appliance_unconfig_funcscope.version >= "5.8":
```

Comment:
I left comment over these commands. 
could you please translate me what happens here  ? :)
 command_set = ('ap', '', '5', '1', '1', 'y', '1', 'n', '0', pwd, TimedCommand(pwd, 45), '')